server/controllers/api/assessmentDayDataController/index.js

```diff
@@ -28,6 +31,13 @@ const AssessmentDayDataController = {
         assessment,
         participant,
       }
+      const assessmentQuery = {
+        name: assessment,
+      }
+      const assessmentVariableQuery = {
+        assessment,
```

Comment:
I think this should be a reference to the assessment record so the relationship doesn't break if the name gets changed.

Reply:
Yup and this does works as a reference. 
In the assessment_variables array the objects within it look like this
```javascript
 {
    assessment: 'assessment_name',
    name: 'variableName'
}```

This makes the assessment property our reference to the assessments collection. I could update the name to `assessment_name` or `assessment_id`.

Reply:
shouldn't the reference be the assessments database ID instead of the name though?

Reply:
Mongo is a NoSQL database and unlike a relational database we shouldn't need to use an ID to make the relations, as a matter of fact a parent document itself can contain child documents which resolves having to make queries using joins and relations as such. 

I'm curious as to why you feel we need to use a BSON ID as the reference, if the client updates the variables of an assessment we can find the variables using the name of the assessment. Also, just a reminder that Lochness generates the assessments, and that's an automated system.